api/src/users/api_views.py

```diff
@@ -73,3 +74,35 @@ class BlockUser(GenericUserExtension):
     @staticmethod
     def field_func(obj_id):
         return CustomUser.objects.get(id=obj_id).blocked_users
+
+
+@method_decorator(name="patch", decorator=swagger_auto_schema(tags=["Users"]))
+@method_decorator(name="put", decorator=swagger_auto_schema(tags=["Users"]))
+class NotificationPreferences(UpdateAPIView):
+    queryset = CustomUser.objects.all()
+    serializer_class = NotificationPreferencesSerializer
+    permission_classes = [
+        IsAuthenticated,
+    ]
+
+    def put_or_patch(self, request, *args, **kwargs):
+        serializer = self.serializer_class(data=request.data)
+
+        # Validate HTTP patch/put data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
+
+        # update the notification preferences field
+        notification_preferences = serializer.data["notification_preferences"]
+        self.request.user.notification_preferences = notification_preferences
+        self.request.user.save()
+
+        return Response(
+            {"notification_preferences": notification_preferences}, status=HTTP_200_OK
```

Comment:
In other parts of the code, we use
`status.HTTP_200_OK`
We might want to be consistent with that too.